src/dataset-builder/ConceptSearch.ts

```diff
@@ -34,15 +35,23 @@ const DebouncedTextInput = withDebouncedChange(TextInput);
 export const ConceptSearch = (props: ConceptSearchProps) => {
   const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
     props;
-  const [search, setSearch] = useState<string>(initialSearch);
+  const [searchText, setSearchText] = useState<string>(initialSearch);
   const [cart, setCart] = useState<Concept[]>(initialCart);
   const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
 
-  useEffect(() => {
+  // when mounting ConceptSearch, we want to retrieve a list of searchConcepts
+  useOnMount(() => {
     void searchConcepts(() => {
-      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root);
     });
-  }, [search, datasetId, domainOption.root, searchConcepts]);
+  });
+  useEffect(() => {
+    if (searchText.length > 2) {
+      void searchConcepts(() => {
+        return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, searchText);
```

Comment:
Before this code had one call to `DataRepo().dataset().searchConcepts()`, now it has two. Is there a way to rewrite this to have only one call instead?

Reply:
Yes, just did!